packages/rulesets/src/spectral/functions/patch-properties-correspond-to-put-properties.ts

```diff
@@ -0,0 +1,65 @@
+// Validates that each patch request body contains one or more properties present in the corresponding put request body,
+// and contains only properties present in the put request body.
+// RPC Code: RPC-Patch-V1-01
+
+import _ from "lodash"
+import { getProperties } from "./utils"
+
+const ERROR_MESSAGE =
+  "A patch request body must only contain properties present in the corresponding put request body, and must contain at least one of the properties."
+const PARAM_IN_BODY = (paramObject: any) => paramObject.in === "body"
+const PATCH = "patch"
+const PUT = "put"
+const PARAMETERS = "parameters"
+
+export const patchPropertiesCorrespondToPutProperties = (pathItem: any, _opts: any, ctx: any) => {
+  if (pathItem === null || typeof pathItem !== "object") {
+    return []
+  }
+
+  const path = ctx.path.concat([PATCH, PARAMETERS])
+  const error = [{ message: ERROR_MESSAGE, path: path }]
+  const errors: any = []
+
+  // array of all the patch body param properties
+  const patchBodyProperties: any[] = pathItem[PATCH]?.parameters?.filter(PARAM_IN_BODY).map((param: any) => getProperties(param.schema))
```

Comment:
>getProperties [](http://example.com/codeflow?start=108&length=13)

Doesnt this just return the top level properties? We possibly need to iterate thru all properties in the patch request to check if the properties specified at each level is also present in the put request body

Reply:
This helper function should get all the properties 

```

 if (schema.allOf && Array.isArray(schema.allOf)) {
    schema.allOf.forEach((base: any) => {
      properties = { ...getProperties(base), ...properties }
    })
  }

```